data-prepper-core/integrationTest.gradle

```diff
@@ -71,54 +75,39 @@ task buildDataPrepperDockerImage(type: DockerBuildImage) {
     images.add("integ-test-pipeline-image")
 }
 
-task createDataPrepper1DockerContainer(type: DockerCreateContainer) {
-    dependsOn buildDataPrepperDockerImage
-    dependsOn createDataPrepperNetwork
-    containerName = "data-prepper1"
-    hostConfig.portBindings = ['21890:21890', '4900:4900']
-    hostConfig.network = createDataPrepperNetwork.getNetworkName()
-    targetImageId buildDataPrepperDockerImage.getImageId()
-}
-
-task createDataPrepper2DockerContainer(type: DockerCreateContainer) {
-    dependsOn buildDataPrepperDockerImage
-    dependsOn createDataPrepperNetwork
-    containerName = "data-prepper2"
-    hostConfig.portBindings = ['21891:21890', '4901:4900']
-    hostConfig.network = createDataPrepperNetwork.getNetworkName()
-    targetImageId buildDataPrepperDockerImage.getImageId()
-}
-
-task startDataPrepper1DockerContainer(type: DockerStartContainer) {
-    dependsOn createDataPrepper1DockerContainer
-    targetContainerId createDataPrepper1DockerContainer.getContainerId()
-    doLast {
-        sleep(10*1000)
+def createDataPrepperDockerContainer(final String taskBaseName, final String dataPrepperName, final int grpcPort,
+                                     final int serverPort, final String pipelineConfigYAML) {
+    return tasks.create("create${taskBaseName}", DockerCreateContainer) {
+        dependsOn buildDataPrepperDockerImage
+        dependsOn createDataPrepperNetwork
+        containerName = dataPrepperName
+        hostConfig.portBindings = ['%d:21890'.formatted(grpcPort), '%d:4900'.formatted(serverPort)]
+        hostConfig.network = createDataPrepperNetwork.getNetworkName()
+        cmd = ["java", "-jar", "data-prepper.jar", pipelineConfigYAML, "/app/data_prepper.yml"]
+        targetImageId buildDataPrepperDockerImage.getImageId()
     }
 }
 
-task startDataPrepper2DockerContainer(type: DockerStartContainer) {
-    dependsOn createDataPrepper2DockerContainer
-    targetContainerId createDataPrepper2DockerContainer.getContainerId()
-    doLast {
-        sleep(10*1000)
+def startDataPrepperDockerContainer(final DockerCreateContainer createDataPrepperDockerContainerTask) {
+    return tasks.create("start${createDataPrepperDockerContainerTask.getName()}", DockerStartContainer) {
+        dependsOn createDataPrepperDockerContainerTask
+        targetContainerId createDataPrepperDockerContainerTask.getContainerId()
+        doLast {
+            sleep(10*1000)
+        }
```

Comment:
nit - I wonder if sleep can be extracted out, thinking about the scenario when we start 2 data preppers causing a 20 second wait total. When really it would be just as effective to start them both simultaneously then wait the 10 seconds.

Reply:
Great catch! I will refactor it out and add it in the e2e task.